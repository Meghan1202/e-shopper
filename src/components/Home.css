.card__row {
    display: flex;
    flex-direction: row;
}

@media only screen and (max-width: 600px) {
  .card__row {
  display: flex;
  flex-direction: column;
  }

} 