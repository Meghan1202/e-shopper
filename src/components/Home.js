import React from 'react';
import Product from './Product';
import './Home.css';
import NavBar from './NavBar';

export default class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      products: [{
        id: 1,
        name: 'Banana',
        price: 40,
        quantity: 0,
      }, {
        id: 2,
        name: 'Apple',
        price: 40,
        quantity: 0,
      }, {
        id: 3,
        name: 'Orange',
        price: 40,
        quantity: 0,
      }, {
        id: 4,
        name: 'Coconut',
        price: 40,
        quantity: 0,
      }, {
        id: 5,
        name: 'Banana',
        price: 40,
        quantity: 0,
      }],
      cartCount: 0,
    };
  }

  onIncrement(id) {
    const { cartCount } = this.state;
    const { products } = this.state;
    const newState = {
      ...this.state,
      cartCount: cartCount + 1,
      products: products.map((eachProduct) => {
        if (eachProduct.id === id) {
          return { ...eachProduct, quantity: eachProduct.quantity + 1 };
        }
        return eachProduct;
      }),
    };
    this.setState(newState);
  }

  onDecrement(id) {
    const { cartCount } = this.state;
    if (cartCount === 0) return;
    const { products } = this.state;
    const newState = {
      ...this.state,
      cartCount: cartCount - 1,
      products: products.map((eachProduct) => {
        if (eachProduct.id === id && eachProduct.quantity > 0) {
          return { ...eachProduct, quantity: eachProduct.quantity - 1 };
        }
        return eachProduct;
      }),
    };
    this.setState(newState);
  }

  render() {
    const { products } = this.state;
    const { cartCount } = this.state;
    return (
      <>
        <NavBar items={cartCount} />
        <h3> Fruits And Vegitables</h3>
        <div className="card__row">
          {products.map((product) => (
            <Product
              key={product.id}
              productId={product.id}
              productName={product.name}
              productPrice={product.price}
              productQuantity={product.quantity}
              onIncrement={() => { this.onIncrement(product.id); }}
              onDecrement={() => { this.onDecrement(product.id); }}
            />
          ))}
        </div>
      </>
    );
  }
}
