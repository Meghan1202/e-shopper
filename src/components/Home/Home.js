import React from 'react';
import PropTypes from 'prop-types';
import Product from '../Product/Product';
import './Home.css';

const Home = ({
  onIncrement, onDecrement, products,
}) => (
  <>
    {console.log(products)}
    <div data-testid="home-page">
      {Object.keys(products).map((category) => (
        <React.Fragment key={category}>
          <h3>{category}</h3>
          <div className="card__row">
            {products[category].map((product) => (
              <Product
                key={product.id}
                productId={product.id}
                productName={product.name}
                productPrice={product.price}
                productQuantity={Number.parseInt(product.stock, 10)}
                count={product.count}
                companyName={product.companyName}
                imgSrc={product.imgSrc}
                onIncrement={() => onIncrement(product.id, product.category)}
                onDecrement={() => onDecrement(product.id, product.category)}
              />
            ))}
          </div>
        </React.Fragment>
      ))}
    </div>
  </>
);

Home.propTypes = {
  products: PropTypes.objectOf(PropTypes.array).isRequired,
  onDecrement: PropTypes.func.isRequired,
  onIncrement: PropTypes.func.isRequired,
};
export default Home;
