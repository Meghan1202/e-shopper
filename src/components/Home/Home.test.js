import React from 'react';
import { render, screen } from '@testing-library/react';
import Home from './Home';

describe(Home.name, () => {
  const mockValue = {
    product: [],
    onIncrement: jest.fn(),
    onDecrement: jest.fn(),
  };
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('should match snapshot for theme light', () => {
    const { product, onIncrement, onDecrement } = mockValue;
    const { container } = render(
      <Home products={product} onIncrement={onIncrement} onDecrement={onDecrement} />,
    );
    expect(container).toMatchSnapshot();
  });
  test('should display the category name', () => {
    const { product, onIncrement, onDecrement } = mockValue;
    render(<Home products={product} onIncrement={onIncrement} onDecrement={onDecrement} />);
    screen.getByText('Fruits And Vegitables');
  });
  test('should render the Product Component', () => {
    const { product, onIncrement, onDecrement } = mockValue;
    render(<Home products={product} onIncrement={onIncrement} onDecrement={onDecrement} />);
    screen.getByTestId('product-card');
  });
});
