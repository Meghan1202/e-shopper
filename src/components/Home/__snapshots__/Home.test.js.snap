// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot for theme light 1`] = `
<div>
  <div
    data-testid="home-page"
  />
</div>
`;
